test/lifecycle_module_test.dart

```diff
@@ -174,218 +222,397 @@ void main() {
-      expect(module.isDisposed, isTrue);
-    });
-
-    test(
-        'should do nothing when the module is unloaded if it was already unloaded',
-        () async {
-      await module.load();
-      await module.unload();
-      module.eventList.clear();
-      await module.unload();
-      expect(module.eventList, equals([]));
-    });
-
-    test('should warn when the module is unloaded if it was never loaded',
-        () async {
-      await module.unload();
-      expect(lastLogMessage, isNotNull);
-      expect(lastLogMessage.level, equals(Level.WARNING));
-    });
-
-    test('should warn when the module is unloaded if it was already unloaded',
-        () async {
-      await module.load();
-      await module.unload();
-      expect(lastLogMessage, isNull);
-      await module.unload();
-      expect(lastLogMessage, isNotNull);
-      expect(lastLogMessage.level, equals(Level.WARNING));
-    });
-
-    test('should throw StateError when attempting to reload a module',
-        () async {
-      await module.load();
-      await module.unload();
-      expect(module.load(), throwsStateError);
-    });
-
-    test(
-        'should trigger suspend events and call onSuspend when module is suspended',
-        () async {
-      await module.load();
-      module.eventList.clear();
-      await module.suspend();
-      expect(
-          module.eventList, equals(['willSuspend', 'onSuspend', 'didSuspend']));
-    });
-
-    test('should set isSuspended when module is suspended', () async {
-      expect(module.isSuspended, isFalse);
-      await module.load();
-      await module.suspend();
-      expect(module.isSuspended, isTrue);
-    });
-
-    test('should do nothing when module is suspended if it was never loaded',
-        () async {
-      await module.suspend();
-      expect(module.eventList, equals([]));
-    });
-
-    test('should warn when module is suspended if it was never loaded',
-        () async {
-      await module.suspend();
-      expect(lastLogMessage, isNotNull);
-      expect(lastLogMessage.level, equals(Level.WARNING));
-    });
-
-    test(
-        'should trigger resume events and call onResume when module is resumed',
-        () async {
-      await module.load();
-      await module.suspend();
-      module.eventList.clear();
-      await module.resume();
-      expect(module.eventList, equals(['willResume', 'onResume', 'didResume']));
-    });
-
-    test('should set isSuspended when module is resumed', () async {
-      await module.load();
-      await module.suspend();
-      expect(module.isSuspended, isTrue);
-      await module.resume();
-      expect(module.isSuspended, isFalse);
-    });
+    void testInvalidTransitions(
+        LifecycleState state, List<LifecycleState> invalidStates) {
+      invalidStates.forEach((fromState) {
+        test('should throw StateError when state is $fromState', () async {
+          await gotoState(module, fromState);
+          expect(module.state, fromState);
```

Comment:
#nit Most of the `expect()` calls in here use the `equals()` matcher, but this does not. I'm fine with either, but we should be consistent